api/src/download-addon.ts

```diff
@@ -0,0 +1,83 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+// For CLI usage
+// npm run download-addon <user/repo> [addonId]
+
+import { fetch } from "undici";
+import config from "./config.js";
+import { STORAGE_FOLDER, createDirForFile } from "./utils/misc.js";
+import path from "path";
+import { writeFile } from "fs/promises";
+
+const ADDONS_FOLDER = STORAGE_FOLDER("addons");
+
+const { token } = config.github;
+
+const headers: Record<string, string> = {};
+if (token) {
+  headers.Authorization = `token ${token}`;
```

Comment:
Quoting GitHub REST documentation:

> `curl -I https://api.github.com --header "Authorization: Bearer YOUR-TOKEN"`

Best to be safe than sorry and just obey by what they advise.

```suggestion
  headers.Authorization = `Bearer ${token}`;
```